src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -100,5 +132,33 @@ public void AddTagHelperAttribute([NotNull] string name, object value)
         {
             AllAttributes.Add(name, value);
         }
+
+        /// <summary>
+        /// Executes the child content asynchronously.
+        /// </summary>
+        /// <returns>A task that indicates when child content execution has completed.</returns>
+        public Task ExecuteChildContentAsync()
+        {
+            return _executeChildContentAsync();
+        }
+
+        /// <summary>
+        /// Execute and retrieve the rendered child content asynchronously.
+        /// </summary>
+        /// <returns>The rendered child content.</returns>
+        /// <remarks>
+        /// Child content is only executed once. Successive calls to this method return a cached result.
+        /// </remarks>
+        public async Task<string> GetChildContentAsync()
+        {
+            if (_childContent == null)
```

Comment:
Maybe use a boolean flag here instead of using `_childContent == null` here to determine if the content was rendered? There's a risk that you could end up in a bad state if the caller does something equivalent to

`await Task.WhenAll(GetChildContentAsync(), GetChildContentAsync())`